docs(docs-infra): Improves symbol linking for Angular Aria selectors

SkyZeroZx opened the pull request:
Improves the symbol linking logic to handle Angular component selectors (e.g., ngCombobox). It attempts to convert Angular selector patterns to their corresponding class names, improving navigation to Angular API documentation.

Fixes #65567

 

## What is the current behavior?
 
<img width="756" height="631" alt="image" src="https://github.com/user-attachments/assets/2b23443e-9523-4743-b86e-a43f69057e66" />
 


## What is the new behavior?

<img width="764" height="597" alt="image" src="https://github.com/user-attachments/assets/a422de46-a4df-4f0b-a861-bd7b22dcb6fe" />

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

JeanMeche commented on adev/shared-docs/pipeline/shared/linking.mts:
```diff
@@ -52,6 +52,27 @@ export function getSymbolUrl(symbol: string, apiEntries: ApiEntries): string | u
+      const moduleName = apiEntries[capitalized];
+      return `/api/${moduleName}/${capitalized}${propName ? `#${propName}` : ''}`;
+    }
+  }
```
This seems really to brittle. 

I was more thiking of changing the structure of `ApiEntries` to `Record<string, string|{moduleName: string, aliasTo?: string}>;`

SkyZeroZx replied:
That makes sense to me, I'll review it to iterate and update it.

JeanMeche commented on adev/shared-docs/pipeline/shared/linking.mts:
```diff
@@ -16,7 +16,8 @@ export function shouldLinkSymbol(symbol: string): boolean {
 
-export type ApiEntries = Record<string, string>; // symbolName -> moduleName (without @angular/ prefix)
+// symbolName -> moduleName (without @angular/ prefix) or symbol info with optional alias
+export type ApiEntries = Record<string, string | {moduleName: string; aliasTo?: string}>;
```
Thinking more about it, should we go with `Record<string, {moduleName: string; aliasTo?: string}>;` ?

SkyZeroZx replied:
Updated: Added `aliases` support for entry points, and updated the type definition:

```ts
export type ApiEntries = Record<
  string,
  {
    moduleName: string;
    targetSymbol?: string;
  }
>;
```
The `targetSymbol` field indicates the symbol associated with previously parsed `aliases`.

JeanMeche commented on adev/src/content/aria/aria-accordion.json:
```diff
@@ -6,6 +6,7 @@
   "entries": [
     {
       "name": "AccordionPanel",
+      "aliasTo": "ngAccordionPanel",
```
This file is auto-generated (and depend on the comiler-cli changes to have been merged, and the components repo to have integrated them) 

But at least it showed that your change worked.

SkyZeroZx commented on packages/compiler-cli/src/ngtsc/docs/src/entities.ts:
```diff
@@ -95,6 +95,7 @@ export interface DocEntryWithSourceInfo extends DocEntry {
 /** Base type for all documentation entities. */
 export interface DocEntry {
   entryType: EntryType;
+  aliasTo?: string;
```
I'm inclined to use `aliases`; `selectors` I think exists, or I seem to have seen it somewhere, and it might be more confusing.

SkyZeroZx committed:
docs(docs-infra): Improves symbol linking for Angular Aria selectors

Improves the symbol linking logic to handle Angular component selectors (e.g., ngCombobox). It attempts to convert Angular selector patterns to their corresponding class names, improving navigation to Angular API documentation.
